File: Geklaut/06_Spring_Wirtschaftstag_Janssen/src/main/java/janssen/app/models/Participation.java
package janssen.app.models;

import lombok.Data;
import javax.persistence.*;

import java.util.List;

@Data
@Entity
public class Participation {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private int id;

  private double price;
  private double paidAlready;
  private String comments;

  @ManyToOne
  private Event at;

  @ManyToOne
  private Responsible responsible;

  @ManyToOne
  private Company company;

  @ManyToMany
  private List<Department> interestedIn;

  @OneToMany
  private List<Timeslot> slots;

}

File: KOTLIN/Wirtschaftstag/WTMobile/app/src/main/java/com/example/wtmobile/data/models/Participation.kt
package com.example.wtmobile.data.models

import java.io.Serializable

class Participation(
    var id: Int,
    var price: Double,
    var paidAlready: Double,
    var comments: String,
    var event: Event,
    var responsible: Responsible,
    var interestedDepartments: List<Department>,
    var company: Company,
    var timeSlotOffers: List<TimeSlot>
) : Serializable
